Annotation

GrigorTumanyan opened the pull request:
Annotation created

GrigorTumanyan committed:
Annotation

GrigorTumanyan committed:
Added CustomerDto class

GrigorTumanyan committed:
update

LevonAloyan commented on src/annotations/validateAnnotations/AdulthoodValidator.java:
```diff
@@ -0,0 +1,31 @@
+public class AdulthoodValidator implements Validator {
+
+    @Override
+    public void fieldValidator(Class<?> className) {
```
Make this class Generic and use generic type as a method parameter 
public class AdulthoodValidator <T> implements Validator {
@Override
    public void fieldValidator(T dto) {

LevonAloyan commented on src/annotations/validateAnnotations/AdulthoodValidator.java:
```diff
@@ -0,0 +1,31 @@
+                    LocalDate fieldValue = (LocalDate) field.get(className.getDeclaredConstructor().newInstance());
+                    if ((LocalDate.now().getYear() - fieldValue.getYear()) <= 18){
+                        if ((LocalDate.now().getDayOfYear() - fieldValue.getDayOfYear()) < 0){
+                            throw new RuntimeException("You are under 18 years old");
```
Instead of throwing exception, create a List<String> messages = new ArrayList<>();
and add into that list message about validation. For example in this case it should be, 'You are under 18 years old".
If Dto did not pass multiple validations then at the end you must have a messages list with multiple elements.

GrigorTumanyan committed:
added generic in classes
change parameter in method fieldValidator
added Message class and message static method for errors

GrigorTumanyan committed:
added generic in classes
change parameter in method fieldValidator
added Message class and message static method for errors
